Scheduler/Controllers/GoalController.cs

```diff
@@ -1,89 +1,117 @@
-﻿using Microsoft.AspNetCore.Mvc;
+﻿using System.Collections.Generic;
+using System.Linq;
+using Microsoft.AspNetCore.Mvc;
 
 namespace Scheduler.Controllers
 {
 	[Route("api/[controller]")]
 	[ApiController]
 	public class GoalController : ControllerBase
 	{
-		[HttpPost]
+		private readonly GoalContext _goalContext;
+
+		public GoalController(GoalContext goalContext)
+		{
+			_goalContext = goalContext;
+		}
+
+		#region Goal CRUD operations
+		[Route("api/[controller]/[action]")]
+		[HttpPost("AddGoal")]
 		public long AddGoal(Goal goal)
 		{
-			using (var goalContext = new GoalContext())
-			{
-				goalContext.Add(goal);
-				goalContext.SaveChanges();
-				return goal.GoalId;
-			}
+			goal.GoalId = 0;
+			_goalContext.Add(goal);
+			_goalContext.SaveChanges();
+
+			return goal.GoalId;
 		}
 
-		[HttpPut]
+		[Route("api/[controller]/[action]")]
+		[HttpPut("UpdateGoal")]
```

Comment:
Here and below: api/[controller] is already in controller attribute. Rewrite this. As I remember you can do this withoute Route attribute